File: stack/stack-with-array.ts
class Stack<T> {
    private items: T[] = [];

    push(value: T): void {
        this.items.push(value);
    }

    pop(): T | undefined {
        if (this.isEmpty())
            return undefined;
        return this.items.pop();
    }

    peek(): T | undefined {
        if (this.isEmpty())
            return undefined;
        return this.items[this.items.length - 1];
    }

    isEmpty(): boolean {
        return this.items.length === 0;
    }

    size(): number {
        return this.items.length;
    }

    print(): void {
        console.log("items:", this.items);
    }
}

const stack = new Stack<number>();
stack.push(42);
stack.push(8);
stack.push(99);
stack.push(31);
stack.print();
console.log("peek:", stack.peek());
console.log("pop:", stack.pop());
console.log("size:", stack.size());
console.log("pop:", stack.pop());
stack.print();

File: stack/stack-with-array.go
package main

import "fmt"

type stack struct {
	items []int
}

func (s *stack) push(value int) {
	s.items = append(s.items, value)
}

func (s *stack) pop() {
	if len(s.items) > 0 {
		s.items = s.items[:len(s.items)-1]
	}
}

func (s stack) peek() {
	if len(s.items) > 0 {
		fmt.Println("peek:", s.items[len(s.items)-1])
	}
}

func (s stack) size() int {
	return len(s.items)
}

func (s stack) print() {
	fmt.Println(s.items)
}

func main() {
	stack := stack{items: []int{}}
	stack.push(42)
	stack.push(8)
	stack.push(99)
	stack.push(31)
	stack.print()
	stack.peek()
	stack.pop()
	stack.pop()
	fmt.Println("size:", stack.size())
	stack.print()
}
